src/content/developers/docs/smart-contracts/security/index.md

```diff
@@ -1,149 +1,246 @@
 
 ## Prerequisites {#prerequisites}
 
-This will cover smart contract security so make sure you're familiar with [smart contracts](/developers/docs/smart-contracts/) before tackling security.
+Make sure you’re familiar with the [fundamentals of smart contract development](/developers/docs/smart-contracts/) before tackling security. 
 
-## How to write more secure smart contract code {#how-to-write-more-secure-smart-contract-code}
+## Guidelines for building secure Ethereum smart contracts {#smart-contract-security-guidelines}
 
-Before launching any code to Mainnet, it is important to take sufficient precaution to protect anything of value your smart contract is entrusted with. In this article, we will discuss a few specific attacks, provide resources to learn about more attack types, and leave you with some basic tooling and best practices to ensure your contracts function correctly and securely.
+### 1. Design proper access controls {#design-proper-access-controls}
 
-## Audits are not a silver bullet {#audits-are-not-a-silver-bullet}
+In smart contracts, functions marked `public` or `external` can be called by any external party, including externally owned accounts (EOAs) and contract accounts. Specifying public visibility for functions is necessary if you want others to interact with your contract (functions marked `private` cannot be called from outside the contract). However, giving every network participant access to contract functions can cause problems, especially if it means anyone can perform sensitive operations (e.g., minting new tokens). 
 
-Years prior, the tooling for writing, compiling, testing, and deploying smart contracts was very immature, leading many projects to write Solidity code in haphazard ways, throw it over a wall to an auditor who would investigate the code to ensure it functions securely and as expected. In 2020, the development processes and tooling that support writing Solidity is significantly better; leveraging these best practices not only ensures your project is easier to manage, it is a vital part of your project's security. An audit at the end of writing your smart contract is no longer sufficient as the only security consideration your project makes. Security starts before you write your first line of smart contract code, **security starts with proper design and development processes**.
+To prevent unauthorized use of smart contract functions, it is necessary to implement secure access controls. Access control mechanisms restrict the ability to use certain functions in a smart contract to arppoved entities, such as accounts responsible for managing the contract. The **Ownable pattern** and **role-based control** are two patterns useful for implementing access control in smart contracts: 
```

Comment:
```suggestion
To prevent unauthorized use of smart contract functions, it is necessary to implement secure access controls. Access control mechanisms restrict the ability to use certain functions in a smart contract to approved entities, such as accounts responsible for managing the contract. The **Ownable pattern** and **role-based control** are two patterns useful for implementing access control in smart contracts: 
```